kafka/src/test/java/com/metamx/tranquility/kafka/KafkaConsumerTest.java

```diff
@@ -0,0 +1,224 @@
+    producer = new KafkaProducer<>(props, new StringSerializer(), new StringSerializer());
+
+    channel = new BlockingChannel(kafka.serverConfig().hostName(),
+                                  kafka.serverConfig().port(),
+                                  BlockingChannel.UseDefaultBufferSize(),
+                                  BlockingChannel.UseDefaultBufferSize(),
+                                  5000
+    );
+    channel.connect();
+
+    consumerProperties = new Properties();
+    consumerProperties.setProperty("group.id", GROUP_ID);
+    consumerProperties.setProperty("zookeeper.connect", zk.getConnectString());
+
+    zkUtils = ZkUtils.apply(zk.getConnectString(), 5000, 5000, false);
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    channel.disconnect();
+    producer.close();
+    kafka.shutdown();
+    zk.close();
+    FileUtils.deleteDirectory(tempDir);
+  }
+
+  private static KafkaConfig getKafkaTestConfig() throws Exception
+  {
+    int port;
+    try (ServerSocket server = new ServerSocket(0)) {
+      port = server.getLocalPort();
+    }
+
+    Properties props = new Properties();
+    props.put("broker.id", "0");
+    props.put("host.name", "localhost");
+    props.put("port", port);
+    props.put("log.dir", tempDir.getPath());
+    props.put("zookeeper.connect", zk.getConnectString());
+    props.put("replica.socket.timeout.ms", "1500");
+    return new KafkaConfig(props);
+  }
+
+  @Test(timeout=15000)
+  public void testStartConsumersNoCommit() throws Exception
+  {
+    final String topic = "testStartConsumersNoCommit";
+    final String schemaMap = String.format("[ {\"topicPattern\":\"%s\", \"spec\":\"twitter-example.spec\"} ]", topic);
+    final int numMessages = 5;
+    final CountDownLatch latch = new CountDownLatch(numMessages);
+
+    AdminUtils.createTopic(zkUtils, topic, 1, 1, new Properties());
+
+    EventWriter mockEventWriter = EasyMock.mock(EventWriter.class);
+    EasyMock.expect(mockEventWriter.send(MESSAGE)).andStubAnswer(
+        new IAnswer<Boolean>()
+        {
+          @Override
+          public Boolean answer() throws Throwable
+          {
+            latch.countDown();
+            return true;
+          }
+        }
+    );
+
+    WriterController mockWriterController = EasyMock.mock(WriterController.class);
+    EasyMock.expect(mockWriterController.getWriter(topic)).andReturn(mockEventWriter).times(numMessages);
+    mockWriterController.stop();
+    mockWriterController.waitForAll();
+    EasyMock.replay(mockWriterController, mockEventWriter);
+
+    // commitMillis is set high enough that the commit thread should not run during the test
+    KafkaConsumer kafkaConsumer = new KafkaConsumer(consumerProperties, schemaMap, 2, 999999, mockWriterController);
+    kafkaConsumer.start();
+
+    Thread.sleep(500); // ugly, but helps give Kafka time to settle before sending messages
```

Comment:
@gianm yeah, was definitely trying to avoid magic sleeps and brittle tests. I can test it some more without the sleep and see if it fails, but I was seeing occasional sporadic failures before because some messages were getting lost